src/lambda_handler.py

```diff
@@ -1,89 +1,231 @@
-        logger.info(
-            'Creating output using "%s" with contents "%s"', character, contents
-        )
-        result["message"] = cowsay.get_output_string(character, contents)
+        if self.cvss_version == "2.0":
+            if self.cvss_score == 10:
+                self.severity = 4
+            elif self.cvss_score >= 7.0:
+                self.severity = 3
+            elif self.cvss_score >= 4.0:
+                self.severity = 2
+            else:
+                self.severity = 1
+        elif self.cvss_version in ["3.0", "3.1"]:
+            if self.cvss_score >= 9.0:
+                self.severity = 4
+            elif self.cvss_score >= 7.0:
+                self.severity = 3
+            elif self.cvss_score >= 4.0:
+                self.severity = 2
+            else:
+                self.severity = 1
+        await super().save(*args, **kwargs)
+
+
+async def process_nvd(cve) -> str:
+    """Add the provided CVE to the database and return its id."""
+    # Fill fields for the CVE documents from the given JSON files
+    try:
+        cve_id = cve["cve"]["CVE_data_meta"]["ID"]
+    except KeyError:
+        # JSON might be malformed, so we'll log what the CVE object looks like
+        # and then raise an error
+        logger.error("CVE object: %s", cve)
+        raise ValueError("JSON does not look like valid NVD CVE data.")
+    # All fields are there but 'ID' field is empty
+    if not cve_id:
+        raise ValueError("CVE ID is empty.")
+
+    # Reject or remove CVEs that don't have baseMetricV2 or baseMetricV3 CVSS data
+    if not any(k in cve["impact"] for k in ["baseMetricV2", "baseMetricV3"]):
+        """If the CVE is in the database, it needs to be deleted from the database"""
 
-    return result
+        outdated_cve_doc = await CVEDoc.find_one(CVEDoc.id == cve_id)
 
+        if outdated_cve_doc:
+            print("x", end="")
+            await outdated_cve_doc.delete()
+            return cve_id
 
-def task_divide(event) -> dict[str, Union[Optional[float], bool]]:
-    """Divide one number by another and provide the result."""
-    result: dict[str, Union[Optional[float], bool]] = {"result": None, "success": True}
-    numerator: str = event.get("numerator", None)
-    denominator: str = event.get("denominator", None)
+        else:
+            print("x", end="")
+            return cve_id
 
-    if denominator is None or numerator is None:
-        error_msg = "Request must include both a numerator and a denominator."
-        logging.error(error_msg)
-        failed_task(result, error_msg)
     else:
+        version = "V3" if "baseMetricV3" in cve["impact"] else "V2"
+        cvss_base_score = cve["impact"]["baseMetric" + version]["cvss" + version][
+            "baseScore"
+        ]
+        cvss_version_temp = cve["impact"]["baseMetric" + version]["cvss" + version][
+            "version"
+        ]
+
+        # Fill document fields with CVE data
+        print(".", end="")
+
         try:
```

Comment:
This `try:` block needs to begin earlier if we want to catch malformed `cve` JSON data:
```suggestion
        try:
            cvss_base_score = cve["impact"]["baseMetric" + version]["cvss" + version][
                "baseScore"
            ]
            cvss_version_temp = cve["impact"]["baseMetric" + version]["cvss" + version][
                "version"
            ]

            # Fill document fields with CVE data
            print(".", end="")
```